### DISCUSSION TITLE
Questions about Magick.NET Native.dll and Temp Directory

### DISCUSSION THREAD

**Discussion by tjohns2866** [dlemstra/Magick.NET#1366] [Help] [answered]
I've been using Magick.NET with Ghostscript for about a year now and recently started encountering some issues that brought up other questions. 

1. How does the Magick.Net.Initialize() work? Looking at the documentation, I think I understand that the initialize is for the XML files only. If anyone is familiar with C#, I had this call in my Page_Load(), so it was running each time. However, I'm not sure if this is necessary, so I removed it. Everything worked fine for a couple weeks until I started getting the type initializer error again.
2. I have the Magick.Native-Q16-x64.dll in the bin/ folder of my project. However, my application errors out and the message is access denied for the C:/Windows/Temp/Magick.Net.net.20.../ folder. I have tried including the xml files in another directory and using SetNativeLibrary() but the Native file still seems to be accessed in the Temp directory and the folder is always created if deleted? Why is that? I have followed the error's suggestions about adding the IUSR/IIS_IUSRS to the Temp directory and that works but only for a short period of time (which this may be a server/permission policy issue).

I did try SetTempDirectory() to no avail because I am guessing the environment variable on the server needs to be added, but I wanted to address the other questions first to hopefully understand how and when the Native.dll is being accessed and if it creates a new Magick.Net.net... folder ever so often which is causing access denied issues being in the Temp directory.

Thanks in advance!

**Comment by dlemstra** [comment]
You probably don't need to call `MagickNET.Initialize()`. This will be done automatically and should also not cause any type initializer errors. And I don't understand what you mean by`Magick.Net.net.20` because that doesn't look like the name of a folder used in this project. What is the exact name of that folder? And are you using the most recent version of this library?

**Reply by tjohns2866** [reply]
Okay, thanks. The folder is Magick.Net.net.20.8.4.0 which I guess is the version I would have installed in 2021 before we started using this in production. Do you think that could be the reason? I know I can do this via Nuget, but I did have to copy the Native.dll to bin/ folder. Will updating the version, update this file as well?

Thanks!

**Reply by dlemstra** [reply, marked as answer]
When you update the Magick.NET package you will also get a new version of the Native library.

**Reply by tjohns2866** [reply]
Thanks for the help!
